middlewares/validators/subscription.ts

```diff
@@ -0,0 +1,16 @@
+import { NextFunction } from "express";
+import { CustomRequest, CustomResponse } from "../../types/global";
+import { emailRegex } from "../../constants/subscription-validator"; 
+import Joi from 'joi';
+
+export const validateSubscribe = (req: CustomRequest, res: CustomResponse, next: NextFunction) => {
```

Comment:
1. Sanitization: Consider sanitizing the input data to prevent potential security vulnerabilities.
2. Asynchronous Validation: If you're dealing with asynchronous validation (e.g., checking if an email already exists in a database), you might need to use promises or async/await to handle the validation process correctly.

Reply:
1. Now we are trimming the data and even checking them in regex as well.
2. In our current implementation, we handle the check for whether a user is already subscribed in the controller itself. This prevents users from subscribing again with the same email, ensuring proper validation without needing to handle it at the schema level right now.